ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/correction/VersionMatchResolution.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ *  Copyright (c) 2025, 2025 IBM Corporation and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.pde.internal.ui.correction;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.pde.core.plugin.IPluginModelBase;
+import org.eclipse.pde.core.plugin.PluginRegistry;
+import org.eclipse.pde.internal.core.text.bundle.Bundle;
+import org.eclipse.pde.internal.core.text.bundle.BundleModel;
+import org.eclipse.pde.internal.core.text.bundle.RequireBundleHeader;
+import org.eclipse.pde.internal.core.text.bundle.RequireBundleObject;
+import org.eclipse.pde.internal.core.util.VersionUtil;
+import org.eclipse.pde.internal.ui.PDEUIMessages;
+import org.osgi.framework.Constants;
+
+/**
+ * Resolution to add available matching version for Required bundles in MANIFEST
+ */
+public class VersionMatchResolution extends AbstractManifestMarkerResolution {
+	public VersionMatchResolution(int type, IMarker marker) {
+		super(type, marker);
+	}
+
+	@Override
+	protected void createChange(BundleModel model) {
+		String bundleId = marker.getAttribute("bundleId", (String) null); //$NON-NLS-1$
```

Comment:
As this requires the bundleId in any case, it should throw an exception if that information is missing
```suggestion
		String bundleId = Objects.requireNonNull(marker.getAttribute("bundleId", (String) null)); //$NON-NLS-1$
```